### ISSUE TITLE
Testing project creation phase fails due to "unitt-cd already exists"

### ISSUE THREAD

**Issue by renedupont** [opendevstack/ods-core#741]
**Describe the bug**

When running `ods-core/tests>make test` the 'Testing project creation' phase failed because of:

```
Testing project creation
=== RUN   TestCreateODSJenkinsThruShellScripts
--- FAIL: TestCreateODSJenkinsThruShellScripts (1.14s)
    create-cd-jenkins_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateNakedOCPProjectsThruShellScripts
--- FAIL: TestCreateNakedOCPProjectsThruShellScripts (0.85s)
    create-projects_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateProjectThruWebhookProxyJenkinsFile
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Failed
--- FAIL: TestCreateProjectThruWebhookProxyJenkinsFile (156.08s)
    jenkinsfile_test.go:152: Actual jenkins stages from run: ods-corejob-create-project-unitt-master-1 don't match -golden:
        '[
          {
            "stage": "Checkout ods-core",
            "status": "SUCCESS"
          },
          {
            "stage": "Checkout ods-configuration",
            "status": "SUCCESS"
          },
          {
            "stage": "Create OpenShift projects",
            "status": "SUCCESS"
          },
          {
            "stage": "Setup Jenkins in CD project",
            "status": "SUCCESS"
          }
        ]
        '
        -jenkins response:
        '[]
        '
FAIL
FAIL    github.com/opendevstack/ods-core/tests/create-projects  158.585s
FAIL
Makefile:4: recipe for target 'test' failed
make: *** [test] Error 1
```

I guess a colleague of mine has already run this on our cluster, because he mentioned that before and it was my first time running that phase. 

It would be nice to be able to run that more than once.

Also, I don't understand that part in the error message, thus I'm not sure if it is caused by the same issue or not:
```
--- FAIL: TestCreateProjectThruWebhookProxyJenkinsFile (156.08s)
    jenkinsfile_test.go:152: Actual jenkins stages from run: ods-corejob-create-project-unitt-master-1 don't match -golden:
...
```

fyi @michaelsauter @georgfedermann @clemensutschig

**Comment by clemensutschig** [comment]
@renedupont the Unit* is removed before every Test... very interesting ist still there.. because the cleanup calls seem to succeed... hmmm

**Comment by clemensutschig** [comment]
https://www.mankier.com/1/oc-delete

We should include ..
--now=true and --wait=true ... this will ensure that the cleanup really happens and we are Good to Go.

Same applies for the jenkins job..

**Comment by renedupont** [comment]
I thought about it again, and this issue might only happen in a theoretical case that I manually established:
I deleted the ODSVERIFY project manually to surpass the verify phase (due to this bug #740) but did not delete the unitt project. By default provision.cleanup.incomplete.projects is set to false and therefore it did not delete it. 
I'm not 100% sure yet, I have to test first but this might be a consequential issue of #740.
